karyon3-core/src/main/java/com/netflix/karyon/http/UriPatternType.java

```diff
@@ -0,0 +1,152 @@
+package com.netflix.karyon.http;
+
+/**
+ * Copyright (C) 2008 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * An enumeration of the available URI-pattern matching styles
+ * 
+ * @since 3.0
+ */
+public enum UriPatternType {
+    SERVLET, REGEX;
+
+    public static UriPatternMatcher get(UriPatternType type, String pattern) {
+        switch (type) {
+        case SERVLET:
+            return new ServletStyleUriPatternMatcher(pattern);
+        case REGEX:
+            return new RegexUriPatternMatcher(pattern);
+        default:
+            return null;
+        }
+    }
+
+    public static String getUri(String uri) {
+        // Strip out the query, if it existed in the URI. See issue 379.
+        int queryIdx = uri.indexOf('?');
+        if (queryIdx != -1) {
+            uri = uri.substring(0, queryIdx);
+        }
+        return uri;
+    }
+
+    /**
+     * Matches URIs using the pattern grammar of the Servlet API and web.xml.
+     *
+     * @author dhanji@gmail.com (Dhanji R. Prasanna)
+     */
+    private static class ServletStyleUriPatternMatcher implements
+            UriPatternMatcher {
+        private final String pattern;
+        private final Kind patternKind;
+
+        private static enum Kind {
+            PREFIX, SUFFIX, LITERAL,
+        }
+
+        public ServletStyleUriPatternMatcher(String pattern) {
+            if (pattern.startsWith("*")) {
+                this.pattern = pattern.substring(1);
+                this.patternKind = Kind.PREFIX;
```

Comment:
The naming seems backwards here. If the pattern starts with an `*` then the suffix is what is important.